packages/riipen-ui-docs/src/pages/components/popover/Demo.jsx

```diff
@@ -0,0 +1,217 @@
+import React from "react";
+
+import Popover from "@riipen-ui/components/Popover";
+import Button from "@riipen-ui/components/Button";
+import Radio from "@riipen-ui/components/Radio";
+import RadioGroup from "@riipen-ui/components/RadioGroup";
+import Form from "@riipen-ui/components/Form";
+
+export default function Simple() {
+  const [state, setState] = React.useState({
+    anchorEl: null,
+    isOpen: false,
+    anchorPosition: {
+      vertical: "top",
+      horizontal: "left"
+    },
+    contentPosition: {
+      vertical: "top",
+      horizontal: "left"
+    }
+  });
+
+  const buttonClick = () => {
+    setState({
+      ...state,
+      isOpen: !state.isOpen
+    });
+  };
+
+  const handleClose = () => {
+    setState({
+      ...state,
+      isOpen: false
+    });
+  };
+
+  const handleAnchorVertical = (e, value) => {
+    setState({
+      ...state,
```

Comment:
I believe you don't have to explicitly merge state. setState should do a shallow merge of top level state!

Reply:
https://stackoverflow.com/a/55342704/8105193
doesnt appear to be so, and I was having the issue where the full state was clearing on a setState call

Reply:
Yup you are right, when I saw setState I was thinking of class based components 🤦‍♂ !

I believe the way `useState` is intended to be used is to hold/update one piece of state per hook usage. For example you could have anchorPosition as one piece of state. This article is a good primer: https://blog.logrocket.com/a-guide-to-usestate-in-react-ecb9952e406c/

That way you don't care what the rest of the state is!